psdisk/psdisk.c

```diff
@@ -171,50 +585,130 @@ static int psdisk_createImg(int optind, char *argv[])
 
 static void psdisk_destroy(void)
 {
+	struct plist_node_t *node;
+
+	while (!LIST_EMPTY(&psdisk_common.list)) {
+			node = LIST_FIRST(&psdisk_common.list);
+			LIST_REMOVE(node, ptrs);
+			free(node);
+	}
+
 	if (psdisk_common.file != NULL)
 		fclose(psdisk_common.file);
 }
 
 
+static int psdisk_handleImg(void)
+{
+	int res = -1;
+	int rmFile = 0;
+
+	if (CREATE_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_createImg()) < 0) {
+			printf("Cannot create file system image '%s'.\n", psdisk_common.fileName);
+			rmFile = 1;
+		}
+		else {
+			printf("File system image '%s' was created correctly.\n", psdisk_common.fileName);
+		}
+	}
+	else if (UPDATE_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_updateImg()) < 0)
+			printf("Cannot update file system image '%s'.\n", psdisk_common.fileName);
+		else
+			printf("File system image '%s' was updated correctly.\n", psdisk_common.fileName);
+
+	}
+	else if (READ_IMG(psdisk_common.opts)) {
+		if ((res = psdisk_readPartsTable()) < 0)
+			printf("Cannot read file system image '%s'.\n", psdisk_common.fileName);
+		else
+			psdisk_showPartsTable();
+	}
+	else {
+		printf("Inappropriate option, read help.\n");
```

Comment:
Maybe printing help would be more helpful than RTFMing poor user ;)

Reply:
In the previous review @nalajcie suggested to create interface common for commandline tools. I took into account programs like fdisk, cmake, git and there are RTFM. It also makes sense, if user writes the wrong options, he/she should be informed that smth is wrong.

Reply:
Yes, I agree that user should be informed. I like the approach of printing help as well, but it's ok as it is.